docs/fsharp-cheatsheet.md

```diff
@@ -404,6 +403,31 @@ Another way of implementing interfaces is to use *object expressions*.
 
 <a name="ActivePatterns"></a>Active Patterns
 ---------------
+*Single-case active patterns*:
+
+    // Basic
+    let (|EmailDomain|) (email: string) =
+        let parts = email.Split '@'
+        parts[1]
+    let (EmailDomain emailDomain) = "yennefer@aretuza.org"  // emailDomain = 'aretuza.org'
+
+    // As Parameters
+    let (|Real|) (x: System.Numerics.Complex) =
+        (x.Real, x.Imaginary)
+    let addReal (Real aa) (Real bb) =  // conversion done in the parameters
```

Comment:
```suggestion
    let addReal (Real (xr, _xi)) (Real (yr, _yi)) =  // conversion done in the parameters
        xr + yr
```
(depends if those concepts have been traversed at this point in the exercise)

Reply:
Thanks. I'll think on this one. It's two concepts; but, I agree, so much power comes from joining concepts that aren't obvious.

Reply:
In general I avoid fst and snd and always deconstruct explicitly like this, ignoring the one I don't want.
The main exception to that is that I'll use `fst` for e.g. `xs |> Seq.map fst`
The main trigger for me suggesting anyting is that aa and bb dont mean anything, so I tend to call two equivalent args (esp where there is not real order to them) x and y
Main reasoning is that pattern matching is something you want to encourage use of as it nudges in good directions for code in general, whereas a sea of fst this snd that rarely ends up producing scannable code.